### ISSUE TITLE
Add support other node images

### ISSUE THREAD

**Issue by ReklatsMasters** [vadimdemedes/trevor#49]
I like [`mhart/alpine-node`](https://hub.docker.com/r/mhart/alpine-node). It's extremely small image ~50mb. Would be better to have option to use this image.

**Comment by sanex3339** [comment]
+++

**Comment by vadimdemedes** [comment]
Will this image work as expected when there are native dependencies (need compilation)?

> Would be better to have option to use this image.

Aside from smaller size, how is it better than the official image?

**Comment by SamVerschueren** [comment]
I'm wondering for a while now, can we just use the same image as Travis does? Is it publicly available for is to use and just use whatever they use?

**Comment by vadimdemedes** [comment]
Yeah, I was thinking about it too. Would solve pretty much all user requests and issues.

**Comment by SamVerschueren** [comment]
@vadimdemedes It looks like the images can be found [here](https://quay.io/organization/travisci). [This](https://docs.travis-ci.com/user/common-build-problems/#Running-a-Container-Based-Docker-Image-Locally) should be some docs, although not sure if we can do anything with it. Have to take a closer look.

**Comment by charlike** [comment]
just drop the suffix `-onbuild`, that's enough. All of the images will be supported - it won't be "breaking" change.

btw, was just curious about docker and implemented it for the sake of training, of course dont have pretty UI ;d

```js
const readFile = util.promisify(fs.readFile)

function getVersions () {
  return readFile('./.travis.yml', 'utf8').then(
    (str) => yaml.safeLoad(str).node_js
  )
}

const opts = { stdio: 'inherit' }

const pass = (version) => () => console.log('Build', version, 'passed')
const fail = (version) => (er) =>
  console.log('Build', version, 'failed', er.message)

const npmTest = (version) =>
  execa('npm', ['test'], opts).then(pass(version), fail(version))

const npmInstall = (version) =>
  execa('npm', ['install'], opts).then(() => npmTest(version))

const nodeInstall = (v) =>
  execa('docker', ['pull', `node:${v}`], opts).then(() => npmInstall(v))

getVersions().then((versions) => pMap(versions, nodeInstall)).catch((er) => {
  console.log('commit failed', er.stderr)
})

```

that's the whole CI core in reality :D

**Comment by charlike** [comment]
btw, when i tried `trevor` few times i thought that it's damn slow (so slow that my archlinux is lagging a bit) because the `-onbuild`, but it seems that's not the reason, because with above implementation it's fast as other images.

The most slow step using Trevor is "building environment", don't know why.

**Comment by vadimdemedes** [comment]
I think `npm install` takes the majority of that time.
